pkg/config/policy.go

```diff
@@ -0,0 +1,249 @@
+func GetPoliciesForOrdererOrg(config cb.Config, consortiumName, orgName string) (map[string]*Policy, error) {
+	org, ok := config.ChannelGroup.Groups[OrdererGroupKey].Groups[orgName]
+	if !ok {
+		return nil, fmt.Errorf("orderer org %s does not exist in channel config", orgName)
+	}
+
+	return getPolicies(org.Policies)
+}
+
+// GetPoliciesForApplication returns a map of policies for application config group.
+func GetPoliciesForApplication(config cb.Config) (map[string]*Policy, error) {
+	application, ok := config.ChannelGroup.Groups[ApplicationGroupKey]
+	if !ok {
+		return nil, errors.New("application missing from config")
+	}
+
+	return getPolicies(application.Policies)
+}
+
+// GetPoliciesForApplicationOrg returns a map of policies for specific application
+// organization.
+func GetPoliciesForApplicationOrg(config cb.Config, orgName string) (map[string]*Policy, error) {
+	orgGroup, ok := config.ChannelGroup.Groups[ApplicationGroupKey].Groups[orgName]
+	if !ok {
+		return nil, fmt.Errorf("application org %s does not exist in channel config", orgName)
+	}
+
+	return getPolicies(orgGroup.Policies)
+}
+
+// getPolicies returns a map of Policy from given map of ConfigPolicy in organization config group.
+func getPolicies(policies map[string]*cb.ConfigPolicy) (map[string]*Policy, error) {
+	if policies == nil {
+		return map[string]*Policy{}, nil
+	}
+
+	p := map[string]*Policy{}
+	for name, policy := range policies {
+		switch policy.Policy.Type {
+		case int32(cb.Policy_IMPLICIT_META):
+			imp := &cb.ImplicitMetaPolicy{}
+			err := proto.Unmarshal(policy.Policy.Value, imp)
+			if err != nil {
+				return nil, err
+			}
+			rule, err := implicitMetaToString(imp)
+			if err != nil {
+				return nil, err
+			}
+
+			p[name] = &Policy{
+				Type: ImplicitMetaPolicyType,
+				Rule: rule,
+			}
+
+		case int32(cb.Policy_SIGNATURE):
+			sp := &cb.SignaturePolicyEnvelope{}
+			err := proto.Unmarshal(policy.Policy.Value, sp)
+			if err != nil {
+				return nil, err
+			}
+			rule, err := signatureMetaToString(sp)
+			if err != nil {
+				return nil, err
+			}
+
+			p[name] = &Policy{
+				Type: SignaturePolicyType,
+				Rule: rule,
+			}
+
+		default:
+			return nil, fmt.Errorf("unknown policy type: %v", policy.Policy.Type)
+		}
+	}
+	return p, nil
+}
+
+// implicitMetaToString converts a *cb.ImplicitMetaPolicy to a string representation.
+func implicitMetaToString(imp *cb.ImplicitMetaPolicy) (string, error) {
+	args := []string{}
+
+	switch imp.Rule {
+	case cb.ImplicitMetaPolicy_ANY:
+		args = append(args, cb.ImplicitMetaPolicy_ANY.String())
+	case cb.ImplicitMetaPolicy_ALL:
+		args = append(args, cb.ImplicitMetaPolicy_ALL.String())
+	case cb.ImplicitMetaPolicy_MAJORITY:
+		args = append(args, cb.ImplicitMetaPolicy_MAJORITY.String())
+	default:
+		return "", fmt.Errorf("unknown implicit meta policy rule type %v", imp.Rule)
+	}
+
+	args = append(args, imp.SubPolicy)
+
+	return strings.Join(args, " "), nil
```

Comment:
nit: we are joining exactly two strings - do we really need to use a slice and join?

Reply:
thought slice.join will be faster. changes applied.